x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityStatusChangeListener.java

```diff
@@ -45,6 +50,15 @@ public synchronized void licenseStateChanged() {
                 logger.warn("Elasticsearch built-in security features are not enabled. Without authentication, your cluster could be " +
                     "accessible to anyone. See https://www.elastic.co/guide/en/elasticsearch/reference/" + Version.CURRENT.major + "." +
                     Version.CURRENT.minor + "/security-minimal-setup.html to enable security.");
+                if (licenseState.getOperationMode().equals(License.OperationMode.BASIC)
+                    || licenseState.getOperationMode().equals(License.OperationMode.TRIAL)) {
+                    deprecationLogger.deprecate(DeprecationCategory.SECURITY, "security_implicitly_disabled",
+                        "The behavior where the value of [xpack.security.enabled] setting defaults to false for " +
+                            licenseState.getOperationMode() + " licenses is deprecated and will be changed in a future version. " +
```

Comment:
The wording on this is hard.

My suggestion is:
 
> The behavior that disables security by default on ${type} licenses is deprecated.   
> A future version of Elasticsearch will change the default value for [xpack.security.enabled] to "true", for all license types.


I'd like to hear from others as well (Any thoughts @lockewritesdocs?)

Reply:
Also, I think we should use 
```
licenseState.getOperationMode().description()
```
so that it says `basic` rather than `BASIC`

But I'm open to different views on that.

Reply:
Thanks, I'll wait for Adam's input and will adjust the message

Reply:
Thanks for the ping @tvernum 👍 I really like your revision.

I'm proposing two modified options that are slightly different, so choose whichever one you think is best. I agree with Tim that rendering the license name in lowercase is the better choice ("basic license" instead of "BASIC license"):

**Option 1**
Disabling security on ${type} licenses by default is deprecated. A later version of Elasticsearch will set [xpack.security.enabled] to "true", enabling security for all licenses by default.

**Option 2**
Disabling security on ${type} licenses by default is deprecated. A later version of Elasticsearch will set [xpack.security.enabled] to "true" for all licenses and enable security by default.

Reply:
Thanks @lockewritesdocs. My concern with "Disabling security on..." makes it sound like this is something the user does/did/configured and we're deprecating that. Whereas it is something that _we_ do ( as in we change our default configuration when the user does nothing ) . WDYT?

Reply:
@jkakavas, we could modify the statement a bit to say something like:

**Option 1**
>This release deprecates disabling security on ${type} licenses by default. 

However, the change might not be in "this release", correct? For example, users could be on 7.15 but this behavior might be introduced in 7.14. Do we want to include the release where this change occurred?

**Option 2**
>The 7.14 release deprecates disabling security on ${type} licenses by default.

If we _really_ don't want to include any information about a specific release, then we could use language that's more similar to Tim's original statement:

**Option 3**
>The default behavior of disabling security on ${type} licenses is deprecated.

Reply:
Thanks both! I went with 

> The default behavior of disabling security on basic licenses is deprecated. A later version of Elasticsearch will set [xpack.security.enabled] to "true" for all licenses and enable security by default.